src/ocean/core/SmartUnion.d

```diff
@@ -220,6 +330,34 @@ private struct SmartUnionIntern ( U )
      **************************************************************************/
 
     Active active;
+
+    /***************************************************************************
+
+        List of active state names
+
+    ***************************************************************************/
+
+    istring[] active_names = member_string_list();
+
+    /***************************************************************************
+
+        CTFE function to generate the list of active state names for union U.
+
+        Returns:
+            a list containing the names of each of the active states of the
+            smart-union (i.e. the names of the fields of U)
+
+    ***************************************************************************/
+
+    static private istring[] member_string_list ( )
```

Comment:
All initializers in D are required to be CTFE-evaluated, not matter if LHS is constant or not. Also it is allowed to implicitly convert any kind of mutability qualifier from RHS to the one required by LHS upon initialization for that exact reason.